src/NServiceBus.Core/Routing/Legacy/ConfigureMSMQDistributor.cs

```diff
@@ -20,6 +21,7 @@ public static void EnlistWithLegacyMSMQDistributor(this EndpointConfiguration co
             config.GetSettings().Set("LegacyDistributor.Address", masterNodeAddress);
             config.GetSettings().Set("LegacyDistributor.ControlAddress", masterNodeControlAddress);
             config.GetSettings().Set("LegacyDistributor.Capacity", capacity);
+            config.DisableFeature<TimeoutManager>();
```

Comment:
We set the address in `StartableEndpoint` depending on if we are enlisted, not weather the message comes from the distributor or not. The local TM is never used then we enlist.